hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java

```diff
@@ -84,10 +105,25 @@ protected void readInternal(ObjectDataInput in) throws IOException {
         if (recordNotNull) {
             int size = in.readInt();
             cacheRecords = createHashMap(size);
+
+            Data[] orderedKeys = new Data[size];
             for (int i = 0; i < size; i++) {
                 Data key = IOUtil.readData(in);
                 CacheRecord record = in.readObject();
-                cacheRecords.put(key, record);
+                orderedKeys[i] = key;
+                cacheRecords.put(key, new WanWrappedCacheRecord(record, true));
+            }
+
+            // RU_COMPAT_5_3
+            try {
```

Comment:
As discussed, there won't be an `EOFException` here - just deserialize `if (in.getVersion().isGreaterOrEqual(V5_4)`.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/24928/commits/12f00318cd7d78ba56a65173931efd2a5d621580